### DISCUSSION TITLE
Back relations for view collections?

### DISCUSSION THREAD

**Discussion by maxhoffmann** [pocketbase/pocketbase#5266] [Q&A] [answered]
Is it possible to make back relations work for view collections?

I’m trying to create an alternative view collection called `questions_public` for a `questions` collection, that doesn’t return all fields from the `questions` table. There is also another table called `proposals` with a `questionId` that references the `questions` table.

**SQL for `questions_public`:**
```SQL
SELECT
id, title, description, created
FROM
questions
```

When I query the original `questions` collection, back relations are expanded for the proposals:
```js
const response = await pb
  .collection("questions") // <--
  .getOne(incomingMessage.value, {
    expand: "proposals_via_questionId",
  });

// includes expand in the response
```

Doing the same for the `questions_public` view collection doesn’t work: 
```js
const response = await pb
  .collection("questions_public") // <--
  .getOne(incomingMessage.value, {
    expand: "proposals_via_questionId",
  });

// no expand in the response
```

Is there a way to make back relations work for view collections?

**Comment by ganigeorgiev** [comment, marked as answer]
It doesn't work because the relation field in the `proposals` collection is pointing to `questions` and not `questions_public` (_even though they may refer to the same records, views are treated as separate collections since they can be practically anything, ex. `select 1 as id`_).

A hacky workaround could be to try to create a view for the `proposals` that will join the `questions_public` as rel field, something like:

```sql
SELECT p.id,
  qp.id as questionId,
  ... other proposals fields...
FROM proposals p
LEFT JOIN questions_public qp on p.questionId = qp.id
```
And then the expand could be: `proposalsView_via_questionId`.

But the above require too much duplication (including the API rules) so I'm not sure if I could recommend it. An easier alternative could be to move your private `questions` fields in another collection that will have its own API rules and not have `questions_public` at all.
Or if this is not an option you could always send a second `getFullList` request to fetch the related `proposals` matching the specific question:

```js
const question = await pb.collection("questions_public").getOne(incomingMessage.value);
const proposals = await pb.collection("proposals").getFullList({
  filter: `questionId="${question.id}"`,
})
```

**Reply by maxhoffmann** [reply]
Wow, thanks for the quick reply. 🙂 I might have to split up the query then, since I also wanted to add additional fields to `questions_public` via SQLite functions. Thank you!
